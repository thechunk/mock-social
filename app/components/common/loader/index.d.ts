interface ILoader {
    loading: boolean,
    style: import('react-native').StyleProp<import('react-native').ViewStyle>
}
