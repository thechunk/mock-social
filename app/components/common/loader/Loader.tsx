import * as React from 'react';
import {PureComponent} from "react";
import {ActivityIndicator, ActivityIndicatorProps, View} from "react-native";
import styles from './styles';

export default class Loader extends PureComponent<ILoader & ActivityIndicatorProps> {
    render() {
        return this.props.loading ? (
            <View style={[styles.loader, this.props.style]}>
                <ActivityIndicator size="small" {...this.props as ActivityIndicatorProps} />
            </View>
        ) : null;
    }
}