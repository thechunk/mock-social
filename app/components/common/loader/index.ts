/// <reference path='./index.d.ts' />

export {default as Loader} from './Loader';

