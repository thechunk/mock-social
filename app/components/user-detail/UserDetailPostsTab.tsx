import * as React from 'react';
import {PureComponent} from "react";
import withPostsData from "../../hoc/withPostsData";
import {TitleSubtitleListCell} from "../common/list/index";
import {Animated} from "react-native";
import styles, {inherited} from "./styles";
import {Loader} from "../common/loader/index";
import {Color} from "../../styles/global";

export class UserDetailPostsTab extends PureComponent<IUserDetailPostsTab & IWithPostsDataProps & IWithPostsDataOptionalProps,
    IUserDetailPostsTabState> {
    constructor(props: IUserDetailPostsTab & IWithPostsDataProps & IWithPostsDataOptionalProps) {
        super(props);
        this.state = {opacity: new Animated.Value(0)}
    }

    componentDidUpdate(prevProps: Readonly<IUserDetailPostsTab & IWithPostsDataProps & IWithPostsDataOptionalProps>, prevState: Readonly<{}>, snapshot?: any): void {
        if (prevProps.loading && !this.props.loading) {
            Animated.timing(this.state.opacity, {
                toValue: 1,
                duration: 400
            }).start();
        }
    }

    render() {
        return this.props.loading
            ? <Loader loading={true} color={Color.White} style={{justifyContent: 'flex-start'}} />
            : (
                <Animated.View
                    style={[styles.postTabContainer, {opacity: this.state.opacity}]}>
                    {this.props.posts.map(v => (
                        <TitleSubtitleListCell
                            key={v.id}
                            title={v.title}
                            subtitle={v.body}
                            cellStyle={[inherited.cell, styles.postCell]}
                            titleStyle={inherited.title}
                            subtitleStyle={inherited.subtitle}/>
                    ))}
                </Animated.View>
            );
    }
}

export default withPostsData(UserDetailPostsTab);
