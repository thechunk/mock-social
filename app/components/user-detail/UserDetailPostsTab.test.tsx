import * as React from 'react';
import {UserDetailPostsTab} from './UserDetailPostsTab';

import {shallow} from 'enzyme';
import {Loader} from "../common/loader/index";
import {Animated} from "react-native";

describe('UserDetailPostsTab', () => {
    const posts = [{id: 1, title: 'title', body: 'body'}];

    it('renders correctly', () => {
        expect(shallow(<UserDetailPostsTab id={1} posts={posts} />)).toMatchSnapshot();
    });

    it('shows loader', () => {
        const wrapper = shallow(<UserDetailPostsTab id={1} loading={true} />);
        expect(wrapper.find(Loader)).toHaveLength(1);
        expect(wrapper.find(Animated.View)).toHaveLength(0);
    });

    it('shows content', () => {
        const wrapper = shallow(<UserDetailPostsTab id={1} loading={false} posts={posts} />);
        expect(wrapper.find(Loader)).toHaveLength(0);
        expect(wrapper.find(Animated.View)).toHaveLength(1);
    })
});
