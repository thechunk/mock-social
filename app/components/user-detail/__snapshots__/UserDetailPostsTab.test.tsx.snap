// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailPostsTab renders correctly 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "borderTopColor": "rgba(0, 0, 0, 0.2)",
        "borderTopWidth": 0.5,
        "marginTop": 12,
      },
      Object {
        "opacity": 0,
      },
    ]
  }
>
  <TitleSubtitleListCell
    cellStyle={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "minHeight": 48,
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        Object {
          "backgroundColor": "#fff",
          "padding": 12,
        },
      ]
    }
    key="1"
    subtitle="body"
    subtitleStyle={
      Object {
        "fontSize": 12,
      }
    }
    title="title"
    titleStyle={
      Object {
        "fontSize": 16,
      }
    }
  />
</AnimatedComponent>
`;
